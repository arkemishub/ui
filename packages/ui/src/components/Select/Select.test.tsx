/**
 * Copyright 2023 Arkemis S.r.l.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Select } from "./index";
import { act, render } from "@testing-library/react";
import { userEvent } from "@storybook/testing-library";

const mockValues = [
  { id: "1", name: "Test 1" },
  { id: "2", name: "Test 2" },
  { id: "3", name: "Test 3" },
];

describe("Select", () => {
  test("should match snapshot", () => {
    const { asFragment } = render(
      <Select
        values={mockValues}
        onChange={() => null}
        renderLabel={(val) => val.name}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render select", () => {
    const { getByTestId } = render(
      <Select
        values={mockValues}
        onChange={() => null}
        renderLabel={(val) => val.name}
      />
    );
    expect(getByTestId("arke-select")).toBeInTheDocument();
  });

  test("should render values", () => {
    const { getByTestId, getByText } = render(
      <Select
        values={mockValues}
        onChange={() => null}
        renderLabel={(val) => val.name}
      />
    );
    act(() => {
      userEvent.click(getByTestId("arke-select"));
    });
    expect(getByText("Test 1")).toBeInTheDocument();
    expect(getByText("Test 2")).toBeInTheDocument();
    expect(getByText("Test 3")).toBeInTheDocument();
  });

  test("should call onChange when value is selected", () => {
    const onChange = jest.fn();
    const { getByTestId, getByText } = render(
      <Select
        values={mockValues}
        onChange={onChange}
        renderLabel={(val) => val.name}
      />
    );
    act(() => {
      userEvent.click(getByTestId("arke-select"));
    });
    act(() => {
      userEvent.click(getByText("Test 1"));
    });
    expect(onChange).toHaveBeenCalled();
  });

  test("should render selected value", () => {
    const { getByText } = render(
      <Select
        values={mockValues}
        onChange={() => null}
        renderLabel={(val) => val.name}
        value={mockValues[0]}
      />
    );
    expect(getByText("Test 1")).toBeVisible();
  });
});
