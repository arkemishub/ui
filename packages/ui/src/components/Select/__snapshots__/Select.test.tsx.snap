// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <div
      class="relative flex items-center"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        class="select__input"
        data-headlessui-state=""
        data-testid="arke-select"
        id="headlessui-listbox-button-:r0:"
        type="button"
      />
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        class="select__endAdornment"
        data-headlessui-state=""
        id="headlessui-listbox-button-:r1:"
        type="button"
      >
        <svg
          class="select__endAdornment__icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 15L12 18.75 15.75 15m-7.5-6L12 5.25 15.75 9"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
